### PR TITLE
add date rolling

### PR DESCRIPTION
**Pull request by paaragon**
## Informations

Type | Status | Migration
---|---|---
Date roller appender | Ready | Yes

****

## Description
New config param that specify the date pattern to rotate the log file. It switchs between size rotation or date rotation.

## Usage example
Example to use your feature and to improve the documentation after merging your PR:
```
import { Logger } from 'ts-log-debug';

export const logger = new Logger('Log Example');
logger.appenders
  .set('file', {
    type: 'file',
    filename: `${__dirname}/../logs/myfile.log`,
    pattern: '.yyyy-MM-dd',
  });

```

## Todos

- [ ] Tests
- [ ] Coverage
- [ ] Example
- [ ] Documentation

### CODE REVIEW

File: src/appenders/components/FileAppender.ts [hunk 1]
```diff
@@ -127,8 +127,14 @@ export class FileAppender extends BaseAppender {
...
    * @returns {streams.RollingFileStream}
    */
-  private openTheStream(file: string, fileSize: number | undefined, numFiles: number, options: any) {
-    const stream = new streams.RollingFileStream(file, fileSize, numFiles, options);
+  private openTheStream(file: string, fileSize: number | undefined, numFiles: number, pattern: string | undefined, options: any) {
+    let stream = null;
+    if (pattern) {
+      stream = new streams.DateRollingFileStream(file, pattern, options);
+    }
+    {
```
Comment (Romakita) [inline src/appenders/components/FileAppender.ts]: Else is missing no ?

**Comment by Romakita** [comment]
Hello @paaragon 
Thanks for your PR. 
Can you rebase your branch please. I've change the repository structure. All source are located now on packages/logger. Normally, if you use rebase command it should keep and move your change at the right place. Let me know if you have any problem :)

See you
Romain

File: src/appenders/components/FileAppender.ts (hunk 1 above)
Comment (paaragon) [reply src/appenders/components/FileAppender.ts]: My bad, I'll fix it and push into my repo.

**Comment by paaragon** [comment]
I saw the travis job failed with:
```
Bad response: 422 {"message":"Couldn't find a repository matching this job.","error":true}
error Command failed with exit code 1.
```

I don't know what is wrong. Could you help me please?

**Comment by Romakita** [comment]
I'll check that :)

**Comment by Romakita** [comment]
Ok don't care about the message. It the coveralls issue. I change the repository name recently and your branch is based on the latest version. Coveralls try to request the old one.

I can't merge your code. You have to rebase your code from my production branch.

Follow theses commands:

```
git remote add stream https://github.com/TypedProject/logger
git fetch stream
git rebase <yourbranch> stream/production
// resolve your conflict with you tool
git push -f
```

See you :)

**Pull request merged**